lib/Doctrine/ORM/Mapping/Driver/AttributeReader.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping\Driver;
+
+use Attribute;
+use Doctrine\ORM\Mapping\Annotation;
+use ReflectionAttribute;
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionProperty;
+
+use function count;
+use function is_subclass_of;
+
+class AttributeReader
```

Comment:
I haven't grasped these changes fully yet, but will this class be useful to ODM? Or is it only an implementation detail and convenient way for ORM to read annotations?

Reply:
It depends on how much work you want to put into `AttributeDriver`. I copied the code over from `AnnotationDriver` and the way it was setup made it more convenient to use this helper class here. Essentially a smaller version of this can be embedded into `AttributeDriver` after some more refactoring. For example the fact if an attribute is repeatable or not is known, so the driver code could be changed to be explicit.